usr/sbin/p11sak/p11sak.c

```diff
@@ -26,6 +33,54 @@ static const char *default_pkcs11lib = "libopencryptoki.so";
 static void *pkcs11lib = NULL;
 static CK_FUNCTION_LIST *funcs = NULL;
 
+static struct ConfigBaseNode *cfg = NULL;
+
+static void error_hook(int line, int col, const char *msg)
+{
+  fprintf(stderr, "PARSE ERROR: %d:%d: %s\n", line, col, msg);
+}
+
+void dump_attr(CK_ATTRIBUTE_PTR a)
+{
+    const char *typestr;
+    typestr = p11_get_cka(a->type);
+
+    switch (a->ulValueLen) {
+    case 0:
+        printf("          %s: no value\n", typestr);
+        break;
+    case 1:
+        printf("          %s: len=%lu value=0x%02hhx\n",
+               typestr, a->ulValueLen, *((uint8_t *)(a->pValue)));
+        break;
+    case 2:
+        printf("          %s: len=%lu value=0x%04hx\n",
+               typestr, a->ulValueLen, *((uint16_t *)(a->pValue)));
+        break;
+    case 4:
+        printf("          %s: len=%lu value=0x%08x\n",
+               typestr, a->ulValueLen, *((uint32_t *)(a->pValue)));
+        break;
+    case 8:
+        printf("          %s: len=%lu value=0x%016lx\n",
+               typestr, a->ulValueLen, *((uint64_t *)(a->pValue)));
+        break;
+    default:
+        printf("          %s: len=%lu value:", typestr, a->ulValueLen);
+        unsigned char *p = a->pValue;
+        for (unsigned int z = 0; z < a->ulValueLen; z++) {
```

Comment:
Not sure if we want variable declaration (`unsigned int z = 0;`) within the for statement.... Doesn't this require a certain C level? This may cause problems when compiling for different distros.....
Same for unsigned `char *p = a->pValue;` with the default case, but without `{ .. }`.
I guess the C level we use for OCK only allows variable declaration at the beginning of a code block `{ .. }`.

Reply:
Done